### PR TITLE
Refactor Child.main to handle EOF

### PR DESCRIPTION
**Pull request by blueyed**
I've seen child processes being left over after quitting Neovim, looping
in `read()`.
This appears to fix it.

Maybe this is what https://github.com/Shougo/deoplete.nvim/pull/625
tried to address also?!

### CODE REVIEW

File: rplugin/python3/deoplete/child.py [hunk 1]
```diff
@@ -49,31 +49,36 @@ def __init__(self, vim):
...
-            elif name == 'on_event':
-                self._on_event(args[0])
-            elif name == 'merge_results':
-                self._write(self._merge_results(args[0], queue_id))
-
-    def _read(self):
-        self._unpacker.feed(sys.stdin.buffer.read(1))
-        return self._unpacker
+        while True:
+            feed = sys.stdin.buffer.raw.read(102400)
```
Comment (blueyed) [inline rplugin/python3/deoplete/child.py]: This also reads up to 100kb (I am seeing ~10kb at minimum here).
Reading this byte per byte seems to be very wasteful.

**Comment by Shougo** [comment]
It is very big change.
It needs to be testing.

I will test it later.

**Comment by Shougo** [comment]
More testers are needed.

**Comment by lambdalisue** [comment]
I haven't test yet but code looks quite simple and efficient. LGTM

**Comment by Shougo** [comment]
I will test it later.
Please wait.

**Comment by Shougo** [comment]
It seems works.

I will merge it later.

**Comment by Shougo** [comment]
Merged.

**Pull request merged**